Makefile

```diff
@@ -6,6 +6,47 @@ include tests/robustness/makefile.mk
 build:
 	GO_BUILD_FLAGS="${GO_BUILD_FLAGS} -v -mod=readonly" ./scripts/build.sh
 
+.PHONY: build-linux-amd64
+build-linux-amd64:
+	GOOS=linux GOARCH=amd64 GO_BUILD_FLAGS="${GO_BUILD_FLAGS} -v -mod=readonly" ./scripts/build.sh
+
+.PHONY: build-linux-386
+build-linux-386:
+	GOOS=linux GOARCH=386 GO_BUILD_FLAGS="${GO_BUILD_FLAGS} -v -mod=readonly" ./scripts/build.sh
+
+.PHONY: build-darwin-amd64
+build-darwin-amd64:
+	GOOS=darwin GOARCH=amd64 GO_BUILD_FLAGS="${GO_BUILD_FLAGS} -v -mod=readonly" ./scripts/build.sh
+
+.PHONY: build-darwin-arm64
+build-darwin-arm64:
+	GOOS=darwin GOARCH=arm64 GO_BUILD_FLAGS="${GO_BUILD_FLAGS} -v -mod=readonly" ./scripts/build.sh
+
+.PHONY: build-windows-amd64
+build-windows-amd64:
+	GOOS=windows GOARCH=amd64 GO_BUILD_FLAGS="${GO_BUILD_FLAGS} -v -mod=readonly" ./scripts/build.sh
+
+.PHONY: build-linux-arm
+build-linux-arm:
+	GOOS=linux GOARCH=arm GO_BUILD_FLAGS="${GO_BUILD_FLAGS} -v -mod=readonly" ./scripts/build.sh
+
+.PHONY: build-linux-arm64
+build-linux-arm64:
+	GOOS=linux GOARCH=arm64 GO_BUILD_FLAGS="${GO_BUILD_FLAGS} -v -mod=readonly" ./scripts/build.sh
+
+.PHONY: build-linux-ppc64le
+build-linux-ppc64le:
+	GOOS=linux GOARCH=ppc64le GO_BUILD_FLAGS="${GO_BUILD_FLAGS} -v -mod=readonly" ./scripts/build.sh
+
+.PHONY: build-linux-s390x
+build-linux-s390x:
+	GOOS=linux GOARCH=s390x GO_BUILD_FLAGS="${GO_BUILD_FLAGS} -v -mod=readonly" ./scripts/build.sh
+
+
+.PHONY: build-all
+build-all: build-linux-amd64 build-linux-386 build-darwin-amd64 build-darwin-arm64 build-windows-amd64 \
+		build-linux-arm build-linux-arm64 build-linux-ppc64le build-linux-s390x
```

Comment:
PLATFORMS=linux-amd64 linux-386 linux-arm linux-arm64 linux-ppc64le linux-s390x darwin-amd64 darwin-arm64 windows-amd64 windows-arm64

.PHONY: build-all
build-all:
	@for platform in $(PLATFORMS); do \
		$(MAKE) build-$${platform}; \
	done

.PHONY: build-%
build-%:
	GOOS=$$(echo $* | cut -d- -f 1) GOARCH=$$(echo $* | cut -d- -f 2) GO_BUILD_FLAGS="${GO_BUILD_FLAGS} -v -mod=readonly" ./scripts/build.sh

Reply:
The approach with explicit targets could use make -j,

the base approach should probably consider something like https://github.com/kubernetes-sigs/kind/blob/bbcc02f84c8efd68d20c3a6585c2f282d7c1b8c9/hack/release/build/cross.sh#L37 to get parallelism

Reply:
I am not sure if we want parallelism...context: https://github.com/kubernetes/test-infra/pull/32672#discussion_r1616735750

Reply:
Serial can be a special case of setting N=1, but local users will probably want an option and I think we'll find this is a lot slower. The matrix on github actions would've been parallel and the prow equivilant would be creating a CI job for each.

Reply:
^^ I agree, it makes sense to have parallelism. 
WDYT @jmhbnz @ahrtr @serathius

Reply:
Overall not a big deal, suggest to follow the `for loop` solution
- building with a loop is much simpler, so it has a little better readability.
- Serial building just takes ~2 minutes, which is far less than other workflow check; so it has no impact on the overall workflow time.